ground/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceViewModel.kt

```diff
@@ -56,16 +56,10 @@ constructor(
         }
       emit(Html.fromHtml(html, 0))
     } catch (e: Throwable) {
-      when (e) {
-        is DataStoreException,
-        is TimeoutCancellationException -> {
-          onGetTosFailure()
-        }
-        else -> {
-          Timber.e("Unexpected error: ${e.message}")
-          onGetTosFailure()
-        }
+      if (e !is TimeoutCancellationException && !e.isPermissionDeniedException()) {
+        Timber.e(e, "Failed to load Terms of Service")
       }
+      onGetTosFailure()
```

Comment:
The error handling shows a generic message "Could not connect, try again later" for all kinds of exceptions. Can we file a bug or a TODO to handle this more elegantly?

Reply:
Filed https://github.com/google/ground-android/issues/2858 to track!